Framework/LiveData/src/Kafka/KafkaHistoStreamDecoder.cpp

```diff
@@ -244,19 +242,18 @@ void KafkaHistoStreamDecoder::initLocalCaches(
     histoBuffer = createBufferWorkspace<DataObjects::Workspace2D>(
         "Workspace2D", static_cast<size_t>(spDetMsg->n_spectra()),
         spDetMsg->spectrum()->data(), spDetMsg->detector_id()->data(), nudet);
+  }
 
-    // Load the instrument if possible but continue if we can't
-    if (!instName.empty()) {
-      loadInstrument<DataObjects::Workspace2D>(instName, histoBuffer,
-                                               jsonGeometry);
-      if (rawMsgBuffer.empty()) {
-        histoBuffer->rebuildSpectraMapping();
-      }
-    } else {
-      g_log.warning(
-          "Empty instrument name received. Continuing without instrument");
+  // Load the instrument if possible but continue if we can't
```

Comment:
Should these changes also go into the event decoder?

Reply:
Aren't these changes in the event decoder as well? (provided we are talking about the same thing)

Reply:
They don't seem to be. The buffer creation code is untouched.

Reply:
They look almost identical (side-by-side) except for the closing brace after the buffer creation `599` in the event decoder and `245` in the histo decoder. I will make that change and resubmit.